### DISCUSSION TITLE
Is this support latest version of panva/node-oidc-provider, which is version 8.x?

### DISCUSSION THREAD

**Discussion by ziad696** [adrianbrs/nest-oidc-provider#20] [Q&A] [answered]
Is this support latest version of panva/node-oidc-provider, which is version 8.x?|

I have test install latest panva/node-oidc-provider, but its cause error since v8.x now is esm only  module

**Comment by adrianbrs** [comment]
Hello @ziad696! A year later I looked at this package again, sorry about that :(
Unfortunately, I tried a few approaches to make this library compatible with `oidc-provider@8` but I couldn't, because this new version depends on many other ESM-only packages, so I couldn't even use a bundler to turn `oidc-provider` back into CJS. As mentioned in this [issue](https://github.com/adrianbrs/nest-oidc-provider/issues/22#issuecomment-1532959667), we need to wait until NestJS starts supporting ESM-only packages, but I think it will take some time, so the only option at the moment is to keep using version 7 of `oidc-provider` :/

**Comment by ziad696** [comment]
Hello, thank you for reply and the explanation, Okay, yes, as currently i stick to version 7 and use your package, help a lot in my project, tyvm

**Reply by adrianbrs** [reply]
Hello @ziad696! [`nest-oidc-provider@2.0.0`](https://github.com/adrianbrs/nest-oidc-provider/releases/tag/2.0.0) now supports the ESM-only version of `oidc-provider@8.x`.
Hope this helps :)

**Comment by adrianbrs** [comment, marked as answer]
[nest-oidc-provider@2.0.0](https://github.com/adrianbrs/nest-oidc-provider/releases/tag/2.0.0) now supports the ESM-only version of `oidc-provider@8.x`.
